### PR TITLE
 scala-runners: init at unstable-2020-02-02

### PR DESCRIPTION
**Pull request by hrhino**
Resubmission of  #84068  (apparently you can't push changes and then reopen a PR on github?)

###### Motivation for this change

It's a good script and useful for testing with pre-release and non-current versions of Scala.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
the title of the pr and commit should be `scala-runners: init at unstable-2020-02-02`

**Comment by hrhino** [comment]
As it now is. Thanks.

File: pkgs/development/compilers/scala-runners/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "dwijnand";
+    rev = "95e03c9f9de0fe0ab61eeb6dea2a364f9d081d31";
+    sha256 = "0mvlc6fxsh5d6gsyak9n3g98g4r061n8pir37jpiqb7z00m9lfrx";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin $out/lib
+    sed -ie "s_\Wcs\W_ ${coursier}/bin/coursier _" scala-runner
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/scala-runners/default.nix:16]:
```suggestion
    sed -ie "s|\Wcs\W|${coursier}/bin/coursier|" scala-runner
```

Comment (hrhino) [reply pkgs/development/compilers/scala-runners/default.nix:16]: `|` are preferred over `_`?

Comment (SuperSandro2000) [reply pkgs/development/compilers/scala-runners/default.nix]: ``|`` or ``/`` are prefered. ``_`` is really unusual for nixpkgs.

**Pull request merged**